synapse/daemon.py

```diff
@@ -246,8 +249,21 @@ def share(self, name, item):
             logger.exception(f'onTeleShare() error for: {name}')
 
     def _onDmonFini(self):
-        for name, cell in self.cells.items():
-            cell.fini()
+        for s in self.listenservers:
+            s.close()
+        for name, share in self.shared.items():
+            if isinstance(share, EventBus):
+                share.fini()
+
+        async def afini():
+            for name, share in self.shared.items():
+                if isinstance(share, s_coro.Fini):
+                    await share.fini()
+
+            for link in self.connectedlinks:
+                await link.fini()
+
+        s_glob.plex.addLoopCoro(afini())
```

Comment:
in a nice real world the daemon is going to get fini'd via a interrupt (sigint/sigterm) and in some of our docker cases, given 15-30 seconds for the process to die before a sigkill is sent.  this async function may not be finished by the time the interpreter shutting down though?  just a bit worrisome from a "will things actually be fini'd?" standpoint.

Reply:
Changed to waiting for all links at the same time.